pkg/hostpath/hostpath.go

```diff
@@ -888,11 +835,13 @@ func (hp *hostPath) Prune(kubeClient kubernetes.Interface) {
 			continue
 		}
 		klog.V(2).Infof("pruner: removed volume file %q with missing pod from disk", file)
-		err = mounter.Unmount(hpv.GetVolPathAnchorDir())
-		if err != nil {
-			klog.V(2).Infof("pruner: issue unmounting for volume %s mount id %s: %s", hpv.GetVolID(), hpv.GetVolPathAnchorDir(), err.Error())
-		} else {
-			klog.V(2).Infof("pruner: successfully unmounted volume %s mount id %s", hpv.GetVolID(), hpv.GetVolPathAnchorDir())
+		if mounter != nil {
```

Comment:
so this is only needed when we are in unit tests?
The mounter here is only instantiated if we are in test mode. And so is the clean-up. I would suggest that we add this logic only in unit tests methods only.

```
err = mounter.Unmount(hpv.GetVolPathAnchorDir())
``` 
we only need a reference to the `hpv` and we could create a `mounter` in the unit test.
wdyt ?

Reply:
see my comment above ... I'll create a mock/test version of the mount interface